nsis/src/nsis/file/NsisExecutable.java

```diff
@@ -4,53 +4,113 @@
 import ghidra.app.util.bin.format.pe.PortableExecutable.SectionLayout;
+import ghidra.program.model.data.DataType;
+import nsis.format.InvalidFormatException;
+import nsis.format.NsisScriptHeader;
 
+/**
+ * 
+ * This class represents a Nsis Executable.
+ *
+ */
 public class NsisExecutable {
 
 	public static final String NAME = "NULLSOFT_SCRIPTABLE_INSTALLER_SYSTEM";
 
 	private FactoryBundledWithBinaryReader reader;
-	private long header_offset;
+	private NsisScriptHeader scriptHeader;
+	private long headerOffset;
 
+	/**
+	 * Use createNsisExecutable to create a Nsis Executable object
+	 */
 	public NsisExecutable() {
 	}
 
+	/**
+	 * Creates and initializes a Nsis Executable object
+	 * 
+	 * @param factory      that will be used to create Nsis Executable
+	 * @param byteProvider object that will permit reading bytes from the file.
+	 *                     The lifespan of the byte provider is controlled by
+	 *                     Ghidra.
+	 * @param layout       object used to load PE executables
+	 * @return The Nsis executable object
+	 * @throws IOException
+	 * @throws InvalidFormatException
+	 */
 	public static NsisExecutable createNsisExecutable(GenericFactory factory,
-			ByteProvider bp, SectionLayout layout) throws IOException {
+			ByteProvider bp, SectionLayout layout)
+			throws IOException, InvalidFormatException {
 		NsisExecutable nsisExecutable = (NsisExecutable) factory
 				.create(NsisExecutable.class);
 		nsisExecutable.initNsisExecutable(factory, bp, layout);
 		return nsisExecutable;
 	}
 
 	private void initNsisExecutable(GenericFactory factory, ByteProvider bp,
-			SectionLayout layout) throws IOException {
-		this.reader = new FactoryBundledWithBinaryReader(factory, bp, true);
-		this.header_offset = -1;
-		findHeaderOffset();
-		// TODO Init sections and headers
-	}
-
-	private void findHeaderOffset() throws IOException {
-		if (this.header_offset == -1) {
-			for (long offset = 0; offset
-					+ NsisConstants.NSIS_MAGIC.length <= reader
-							.length(); offset++) {
-				byte[] content = reader.readByteArray(offset,
-						NsisConstants.NSIS_MAGIC.length);
-				if (Arrays.equals(NsisConstants.NSIS_MAGIC, content)) {
-					this.header_offset = offset;
-					return;
-				}
+			SectionLayout layout) throws IOException, InvalidFormatException {
+		this.reader = new FactoryBundledWithBinaryReader(factory, bp,
+				/* isLittleEndian= */ true);
+		this.headerOffset = findHeaderOffset();
+		initScriptHeader(bp);
+	}
+
+	private long findHeaderOffset() throws IOException, InvalidFormatException {
+		long offset = -1;
+		for (long testOffset = 0; testOffset
+				+ NsisConstants.NSIS_MAGIC.length <= reader
+						.length(); testOffset++) {
+			byte[] content = reader.readByteArray(testOffset,
+					NsisConstants.NSIS_MAGIC.length);
+			if (Arrays.equals(NsisConstants.NSIS_MAGIC, content)) {
+				offset = testOffset;
+				break;
 			}
 		}
+		if (offset == -1) {
```

Comment:
small readability suggestion:

```
long findHeaderOffset() {
  for (.....) {
    if (found magic) {
      return header_offset;
    }
  }
  throw new InvalidFormatException("NSIS magic not found");
}
```

This way you never even need to treat a certain value (-1) as special.